### PR TITLE
feat: New Module `avm/res/network/public-ip-prefix`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/network/public-ip-prefix`, migrated from CARML.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.network.public-ip-prefix](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.network.public-ip-prefix.yml) |

### CODE REVIEW

**Comment by matebarabas** [comment]
@krbar, please don't forget to uncomment the related entry in CODEOWNERS file as part of this PR. Thanks!

File: avm/res/network/public-ip-prefix/tests/e2e/waf-aligned/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
```
Comment (AlexanderSehr) [inline avm/res/network/public-ip-prefix/tests/e2e/waf-aligned/main.test.bicep:38]: not used - in turn the entire dependency & file can be removed

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Note: The readme generation was updated in main

File: avm/res/network/public-ip-prefix/tests/e2e/waf-aligned/main.test.bicep (hunk 1 above)
Comment (krbar) [reply avm/res/network/public-ip-prefix/tests/e2e/waf-aligned/main.test.bicep:38]: Correct, removed, thanks for the suggestion.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**